### PR TITLE
change: [M3-7785] - Improve Dev Tools

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This PR brings non-breaking, minor improvement to the dev tools.

## Changes  🔄
- Add a theme select to persist theme choice while using MSW
- Add a rest button to bring the feature flags values back to their Launch Darkly defaults
- Color the dev tools icon red when MSW is enabled
- Add a link to the preference modal for convenience
- Comment out banners present of EVERY page cluttering the UI. (they can be easily re-enabled in `serverHandlers.ts`
- Update documentation

## Preview 📷
Screenshot 2024-02-22 at 11 49 19

### Verification steps
- Pull local code locally and `yarn up` your application
- Confirm behavior of dev tools menu

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/dev-tools/FeatureFlagTool.tsx [hunk 1]
```diff
@@ -48,6 +49,11 @@ export const FeatureFlagTool = withFeatureFlagProvider(() => {
     setStorage(MOCK_FEATURE_FLAGS_STORAGE_KEY, updatedFlags);
   };
 
+  const resetFlags = () => {
```
Comment (mjac0bs) [inline packages/manager/src/dev-tools/FeatureFlagTool.tsx:52]: Can we add a short function comment to specify that this is resetting to Launch Darkly defaults?

File: packages/manager/src/dev-tools/Preferences.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export const Preferences = () => {
+  return (
+    <>
+      <h4 style={{ marginBottom: 4 }}>Preferences</h4>
+      <a
+        href="/profile/settings?preferenceEditor=true"
+        style={{ color: '#fff' }}
+      >
+        Open preference Modal
```
Comment (mjac0bs) [inline packages/manager/src/dev-tools/Preferences.tsx:12]: I love this. ++

File: packages/manager/.changeset/pr-10220-changed-1708622627827.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/manager": Changed
+---
+
+Improve dev toolsUI ([#10220](https://github.com/linode/manager/pull/10220))
```
Comment (mjac0bs) [inline packages/manager/.changeset/pr-10220-changed-1708622627827.md:5]:
```suggestion
Improve dev tools UI ([#10220](https://github.com/linode/manager/pull/10220))
```

File: packages/manager/src/dev-tools/dev-tools.css [hunk 4]
```diff
@@ -13,12 +13,20 @@
   overflow: auto;
 }
 
+#dev-tools.mswEnabled svg {
+  color: #ff4d4f;
```
Comment (mjac0bs) [inline packages/manager/src/dev-tools/dev-tools.css:17]: Wondering about this color choice. Red feels logical in the sense that it draws a developer's attention, but it also generally signifies "error". I wonder if green would be more appropriate here?

File: packages/manager/src/mocks/serverHandlers.ts [hunk 5]
```diff
@@ -1911,8 +1917,8 @@ export const handlers = [
           outageNotification,
           minorSeverityNotification,
           criticalSeverityNotification,
-          abuseTicket,
-          emailBounce,
+          // abuseTicket,
+          // emailBounce,
```
[the remark this answers is not in the document]
Comment (mjac0bs) [reply packages/manager/src/mocks/serverHandlers.ts:1921]: Agreed that these are annoying. I don't think I'll miss them, but not sure if anyone else on the team has a reason behind why they have historically been kept visible by default.

**Review by mjac0bs — COMMENTED** [review]
This is great, thank you for the improvements!! 

Are you seeing the scrollbar to the side of the MSW icon now too? 
Screenshot 2024-02-22 at 9 34 49 AM

**Comment by abailly-akamai** [comment]
@mjac0bs yes I could see it if I turned on always show scrollbar in my OS
 
Screenshot 2024-02-22 at 13 29 49

I fixed it too!

**Review by mjac0bs — APPROVED** [review]
🛠️ ✅ 🎉 

Screenshot 2024-02-22 at 10 30 39 AM

**Review by bnussman-akamai — COMMENTED** [review]
Thanks for doing this!

I did get a bit confused at first when I tried setting a theme and nothing happened because I had the MSW off. 

I wonder if we should somehow indicate that this theme selector is **only** for when the MSW is on. Or maybe we could make Cloud Manager respect the override even when the MSW if off. 

Screenshot 2024-02-22 at 2 17 13 PM

**Comment by abailly-akamai** [comment]
@bnussman-akamai fair enough - it's in the documentation but I'll add it to the label as well)

**Review by bnussman-akamai — APPROVED** [review]
Only other feature I can think of would be a button that clears **all** local storage and reloads the page. It's not a huge deal we don't have it because it's relatively easy to do in dev tools, but I don't think it would hurt to put in our dev tools.

**Comment by abailly-akamai** [comment]
@bnussman-akamai I'll hold on that item cause it has implications and we'd need to pick and choose what to clear. I'd rather leave that to the developer's discretion. Besides, clearing ALL local storage will log you out

**Comment by bnussman-akamai** [comment]
@abailly-akamai Yaya, that's kind of the point. Clearing local storage is super useful for debugging things like our login redirect logic for example. It won't log you out necessarily. 

I'm cool either way. I've never had a problem opening dev tools to do this

File: packages/manager/src/dev-tools/FeatureFlagTool.tsx [hunk 6]
```diff
@@ -75,6 +84,9 @@ export const FeatureFlagTool = withFeatureFlagProvider(() => {
               </div>
             );
           })}
+          <button onClick={resetFlags} style={{ marginTop: 8 }}>
+            Reset default flags
```
Comment (hana-linode) [inline packages/manager/src/dev-tools/FeatureFlagTool.tsx:88]: Can we reword this to `Reset to LaunchDarkly default flags` for more clarity?
```suggestion
            Reset to LaunchDarkly default flags
```

File: packages/manager/src/dev-tools/dev-tools.tsx [hunk 7]
```diff
@@ -10,12 +11,17 @@ import './dev-tools.css';
...
 
 function install(store: ApplicationStore) {
   function DevTools() {
     return (
-      <div id="dev-tools">
-        <div>🛠</div>
+      <div className={isMSWEnabled ? 'mswEnabled' : ''} id="dev-tools">
+        <div>
+          <Handyman />
+        </div>
```
[the remark this answers is not in the document]
Comment (hana-linode) [reply packages/manager/src/dev-tools/dev-tools.tsx:24]: Can we also/or add text to indicate the MSW is "on"? For accessibility, color should not be the only indicator haha

**Review by hana-linode — COMMENTED** [review]
Looks great, have we also considered what support for toggling object feature flags would look like?

File: packages/manager/src/dev-tools/FeatureFlagTool.tsx (hunk 6 above)
Comment (mjac0bs) [reply packages/manager/src/dev-tools/FeatureFlagTool.tsx:88]: I almost made this same comment, then deleted it, and asked for a code comment, ha. I do still think it would be nice. 

I was thinking of (1) more text requiring further height increase of the dev tools panel and (2) whether we'd want to specify LD if to save some trouble of updating it if we were to switch to another tool in the future - but really that's such an easy change.

Comment (hana-linode) [reply packages/manager/src/dev-tools/FeatureFlagTool.tsx:88]: Same brain cell 🧠 . (1) We can shorthand to LD to save some width space (2) Agreed

**Comment by abailly-akamai** [comment]
> Can we also/or add text to indicate the MSW is "on"? For accessibility, color should not be the only indicator haha

@hana-linode I want to avoid taking extra real estate on the page with the dev tools badge, so I made the expanded MSW `enabled` label color the same green which creates a relationship. I also have a line in the doc that speaks to the status icon color. I think that's good enough for now, we can revisit if the need to provide a text based status is strong. All other items have been addressed!
